docs/04-api-reference/front-commerce-core/react.mdx

```diff
@@ -0,0 +1,197 @@
+---
+title: "React"
+sidebar_position: 3
+description: "This documentation will cover all React dedicated APIs."
+---
+
+<p>{frontMatter.description}</p>
+
+To help you creating the best front-end with React, we've created the some React
+piece of code that you can reuse in your components.
+
+## `useConditionalCallOnMount`
+
+Execute a callback when component is mounted if a condition is met.
+
+```js
+useConditionalCallOnMount(condition, callback);
+```
+
+Arguments:
+
+- `condition`: A `boolean` that specify if the callback must be executed
+- `callback`: A function that must be executed if `condition` is `true`
+
+Example:
+
+```jsx
+import { useConditionalCallOnMount } from "@front-commerce/core/react";
+
+const AcmeComponent = ({ props }) => {
+  const logOnMount = () => {
+    console.log("I am mounted");
+  };
+
+  useConditionalCallOnMount(props.shouldLogOnMounted, logOnMount);
+
+  return <div>Acme Component</div>;
+};
+```
+
+## `useIsMount`
+
+A variable that return `true` or `false` depending of the component mount status
+
+```js
+useIsMount();
+```
+
+Return: `boolean`
+
+Example:
+
+```jsx
+import { useIsMount } from "@front-commerce/core/react";
+
+const AcmeComponent = ({ props }) => {
+  const isMounted = useIsMount();
+
+  return (
+    <div>
+      {isMounted ? "Component is mounted" : "Component isn't mounted yet"}
+    </div>
+  );
+};
+```
+
+## `useItemsValidityCheck`
+
+Return an object to tell the status of validty of an array
+
+```js
+useItemsValidityCheck(items);
+```
+
+Arguments:
+
+- `items`: an array of object that have a property named `valid`
+
+Return
+
+`Object`:
+
+- `allValidityResolved` (`boolean`): all items contains a `valid` property
+- `hasValid` (`boolean`): array contain at least one item with `valid` property
+  with value `true`
+- `allValid` (`boolean`): all items have `valid` property with value `true`
+
+Example:
+
+```jsx
+import { useItemsValidityCheck } from "@front-commerce/core/react";
+
+const items = [
+  {
+    title: "My website is using Front-Commerce",
+    valid: true,
+  },
+  {
+    title: "The answer is 42",
+    valid: true,
+  },
+  {
+    title: "My front-end use PHP",
+    valid: false,
+  },
+];
+
+const AcmeComponent = ({ props }) => {
+  const arrayValidation = useItemsValidityCheck(items);
+
+  return (
+    <div>
+      Validity status : Are all items valid ? {arrayValidation.allValid}
+      Is array have at lease one valid item ? {arrayValidation.hasValid}
+      Are all item passed validity resolution:
+      {arrayValidation.allValidityResolved}
+    </div>
+  );
+};
+```
```

Comment:
**issue:**  I don't think it makes sense to invent an irrelevant example. This API is here for legacy reasons, and used only for adding items to cart/wishlist. I think we'd better get rid of it and I'd rather keep the doc succint.

**suggestion:** remove example, and add maybe a sentence saying that it's an API we don't recommend to use unless there is a good use case for it (or something like that :))

Reply:
fixed